### PR TITLE
handle connectablePosition extension in modification network

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 1]
```diff
@@ -722,25 +723,38 @@ private void applyModifications(ConnectablePosition.Feeder feeder,
 
     private void modifyFeederAdder(ConnectablePositionAdder.FeederAdder<?> feeder,
                                    Object modificationInfos,
+                                   Network network,
                                    List<ReportNode> reports,
                                    int feederNumber) {
+        var connectionName = getConnectionName(modificationInfos, feederNumber);
+        var connectionDirection = getConnectionDirection(modificationInfos, feederNumber);
+        var connectionPosition = getConnectionPosition(modificationInfos, feederNumber);
+        var equipmentId = getEquipmentId(modificationInfos);
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:732]: Here you don't really have an equipmentId, the type is AttributeModification<String>. You should remove the var if it brings fuzziness. Same for other vars

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 2]
```diff
@@ -771,6 +785,27 @@ private String getConnectionFieldName(int feederNumber, String baseFieldName) {
...
+                InjectionModificationInfos::getVoltageLevelId);
+    }
+
+    private AttributeModification<String> getBusOrBusbarSectionId(Object modificationInfos, int feederNumber) {
+        return getConnectionDetail(modificationInfos, feederNumber,
+                BranchModificationInfos::getBusOrBusbarSectionId1, BranchModificationInfos::getBusOrBusbarSectionId2,
+                InjectionModificationInfos::getBusOrBusbarSectionId);
+    }
+
+    private AttributeModification<String> getEquipmentId(Object modificationInfos) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:800]: It's a pity we grapple with Object type all the way. modificationInfos should be BasicEquipmentModificationInfos in processConnectivityPosition() signature

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 3]
```diff
@@ -771,7 +785,28 @@ private String getConnectionFieldName(int feederNumber, String baseFieldName) {
...
+                InjectionModificationInfos::getVoltageLevelId);
+    }
+
+    private AttributeModification<String> getBusOrBusbarSectionId(BasicEquipmentModificationInfos modificationInfos, int feederNumber) {
+        return getConnectionDetail(modificationInfos, feederNumber,
+                BranchModificationInfos::getBusOrBusbarSectionId1, BranchModificationInfos::getBusOrBusbarSectionId2,
+                InjectionModificationInfos::getBusOrBusbarSectionId);
+    }
+
+    private AttributeModification<String> getEquipmentId(BasicEquipmentModificationInfos modificationInfos) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:800]: return AttributeModification.toAttributeModification(modificationInfos.getEquipmentId(), OperationType.SET);

Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:816]: ok

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 4]
```diff
@@ -789,18 +835,30 @@ private String getConnectionPositionField(int feederNumber) {
...
     }
 
-    private AttributeModification<Integer> getConnectionPosition(Object modificationInfos, int feederNumber) {
+    private AttributeModification<Integer> getConnectionPosition(BasicEquipmentModificationInfos modificationInfos, int feederNumber) {
         return getConnectionDetail(modificationInfos, feederNumber,
                 BranchModificationInfos::getConnectionPosition1, BranchModificationInfos::getConnectionPosition2,
                 InjectionModificationInfos::getConnectionPosition);
     }
 
+    private int calculatePosition(AttributeModification<Integer> connectionPosition,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:850]: Why adding this method ? Use direct getPosition() at row 735

Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:850]:
```suggestion
    private int getPosition(AttributeModification<Integer> connectionPosition,
```

**Review by etiennehomer — APPROVED** [review]

**Review by thangqp — APPROVED** [review]
Tested with loadflow case (equipment without extension) => OK

**Pull request merged**